script/dump-symbols.py

```diff
@@ -1,62 +1,88 @@
 
+import argparse
 import os
 import sys
 
-from lib.config import PLATFORM
-from lib.util import electron_gyp, execute, rm_rf
+from lib.config import PLATFORM, enable_verbose_mode, is_verbose_mode
+from lib.gn import gn
+from lib.util import execute, rm_rf
 
+ELECTRON_ROOT = os.path.abspath(os.path.dirname(os.path.dirname(__file__)))
+SOURCE_ROOT = os.path.abspath(os.path.dirname(ELECTRON_ROOT))
+RELEASE_PATH = os.path.join('out', 'Release')
 
-SOURCE_ROOT = os.path.abspath(os.path.dirname(os.path.dirname(__file__)))
-DIST_DIR = os.path.join(SOURCE_ROOT, 'dist')
-OUT_DIR = os.path.join(SOURCE_ROOT, 'out', 'R')
-CHROMIUM_DIR = os.path.join(SOURCE_ROOT, 'vendor', 'download',
-                            'libchromiumcontent', 'static_library')
-
-
-def main(destination):
-  rm_rf(destination)
-  (project_name, product_name) = get_names_from_gyp()
+def main():
+  args = parse_args()
+  if args.verbose:
+    enable_verbose_mode()
+  rm_rf(args.destination)
+  source_root = os.path.abspath(args.source_root)
+  build_path = os.path.join(source_root, args.build_dir)
+  product_name = gn(build_path).args().get_string('electron_product_name')
+  project_name = gn(build_path).args().get_string('electron_project_name')
 
   if PLATFORM in ['darwin', 'linux']:
-    generate_breakpad_symbols = os.path.join(SOURCE_ROOT, 'tools', 'posix',
-                                             'generate_breakpad_symbols.py')
+
     if PLATFORM == 'darwin':
       #macOS has an additional helper app; provide the path to that binary also
-      main_app = os.path.join(OUT_DIR, '{0}.app'.format(product_name),
+      main_app = os.path.join(build_path, '{0}.app'.format(product_name),
                             'Contents', 'MacOS', product_name)
       helper_name = product_name + " Helper"
-      helper_app = os.path.join(OUT_DIR, '{0}.app'.format(helper_name),
+      helper_app = os.path.join(build_path, '{0}.app'.format(helper_name),
                             'Contents', 'MacOS', product_name + " Helper")
       binaries = [main_app, helper_app]
+      for binary in binaries:        
+        generate_posix_symbols(binary, source_root, build_path, 
+                                        args.destination)
     else:
-      binaries = [os.path.join(OUT_DIR, project_name)]
-    args = [
-      '--build-dir={0}'.format(OUT_DIR),
-      '--symbols-dir={0}'.format(destination),
-      '--libchromiumcontent-dir={0}'.format(CHROMIUM_DIR),
-      '--clear',
-      '--jobs=16',
-    ]
-    for binary in binaries:
-      args += '--binary={0}'.format(binary),
+      binary = os.path.join(build_path, project_name)
+      generate_posix_symbols(binary, source_root, build_path, 
+                                      args.destination)
 
   else:
-    generate_breakpad_symbols = os.path.join(SOURCE_ROOT, 'tools', 'win',
+    generate_breakpad_symbols = os.path.join(ELECTRON_ROOT, 'tools', 'win',
                                              'generate_breakpad_symbols.py')
```

Comment:
Do we want to use the `generate_breakpad_symbols.py` from Chromium now that we're guaranteed to have it? it's at `//components/crash/content/tools/generate_breakpad_symbols.py`